src/Http/Http.Features/src/FeatureCollection.cs

```diff
@@ -100,7 +100,7 @@ public TFeature Get<TFeature>()
             return (TFeature)this[typeof(TFeature)];
         }
 
-        public void Set<TFeature>(TFeature instance)
+        public void Set<TFeature>([AllowNull] TFeature instance)
```

Comment:
@JamesNK while the compiler used for `dotnet build` supports `TFeature?`, VS 16.8 P2 is required to get the correct design time experience. I'm going to hold off on updating this until that becomes available.